### ISSUE TITLE
missing data.h

### ISSUE THREAD

**Issue by Laci1953** [donnierussellii/zork1c#4]
It seems data.h file is missing

**Comment by donnierussellii** [comment]
Some source files are generated by compress_*.c

**Comment by Laci1953** [comment]
Thanks,
That's clever...
I remember playing Zork 30 years ago, on CP/M... I will try to port your version on CP/M again (Z80).
regards,Ladislau
    On Thursday, November 18, 2021, 12:52:39 AM GMT+2, donnierussellii ***@***.***> wrote:  
 
 


Some source files are generated by compress_*.c

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Laci1953** [comment]
Hi,I ported the game to a version of RC2014 provided with a 512KB RAM + 512KB ROM module.See https://groups.google.com/g/rc2014-z80/c/IDuDXm9FC08
regards,Ladislau

**Comment by donnierussellii** [comment]
I wasn't familiar with this computer, but I love these homebrew machines. It sounds like you enjoy a challenge, working within a limited space.

I set a goal of running Zork I on a single chip, the ATmega2560, with no VM or external loading (from an sdcard, for example). Even the TV composite output and IR input are handled on the chip, using half of the available flash and RAM.

**Comment by Laci1953** [comment]
Hi Donnie,
yes, you're right, I really enjoyed the challenge!
The final result is spectacular: on a Z80 based computer, the full game runs! 
It is true that I "sliced" the source code to fit in 16KB pieces, stored on a 512KB EPROM, but I did not dropped a single line!
Congratulations for your parser, I like-it!
I was using the HiTech C compiler for Z80, which I consider the best available.But, to my surprise, I had some problems related to the HiTech C compiler:
They were 2 big issues:1. strchr(0) does not work (it stops the search BEFORE the final 0, so it will always return 0)2. memset is wrongly implemented (the last two parameters are inversed!!!)
If you will ever use HiTech to port your code to a new environment, keep an eye on those issues!
regards,Ladislau
    On Wednesday, December 15, 2021, 03:21:42 AM GMT+2, donnierussellii ***@***.***> wrote:  
 
 


I wasn't familiar with this computer, but I love these homebrew machines. It sounds like you enjoy a challenge, working within a limited space.

I set a goal of running Zork I on a single chip, the ATmega2560, with no VM or external loading (from an sdcard, for example). Even the TV composite output and IR input are handled on the chip, using half of the available flash and RAM.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by Laci1953** [comment]
Hi,
I published ZORK for RC2014 ( https://groups.google.com/g/rc2014-z80/c/IDuDXm9FC08 ) 
Ladislau



    On Wednesday, December 15, 2021, 08:26:55 AM GMT+2, Ladislau Szilagyi ***@***.***> wrote:  
 
  Hi Donnie,
yes, you're right, I really enjoyed the challenge!
The final result is spectacular: on a Z80 based computer, the full game runs! 
It is true that I "sliced" the source code to fit in 16KB pieces, stored on a 512KB EPROM, but I did not dropped a single line!
Congratulations for your parser, I like-it!
I was using the HiTech C compiler for Z80, which I consider the best available.But, to my surprise, I had some problems related to the HiTech C compiler:
They were 2 big issues:1. strchr(0) does not work (it stops the search BEFORE the final 0, so it will always return 0)2. memset is wrongly implemented (the last two parameters are inversed!!!)
If you will ever use HiTech to port your code to a new environment, keep an eye on those issues!
regards,Ladislau
    On Wednesday, December 15, 2021, 03:21:42 AM GMT+2, donnierussellii ***@***.***> wrote:  
 
 


I wasn't familiar with this computer, but I love these homebrew machines. It sounds like you enjoy a challenge, working within a limited space.

I set a goal of running Zork I on a single chip, the ATmega2560, with no VM or external loading (from an sdcard, for example). Even the TV composite output and IR input are handled on the chip, using half of the available flash and RAM.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
